Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -7935,20 +7951,77 @@ void WebPage::completeTextManipulation(const Vector<WebCore::TextManipulationIte
         return;
     }
 
-    auto* localMainFrame = dynamicDowncast<LocalFrame>(m_page->mainFrame());
-    RefPtr mainDocument = localMainFrame ? localMainFrame->document() : nullptr;
-    if (!mainDocument) {
+    if (items.isEmpty()) {
         completionHandler(true, { });
         return;
     }
 
-    auto* controller = mainDocument->textManipulationControllerIfExists();
-    if (!controller) {
-        completionHandler(true, { });
+    auto currentFrameID = items[0].frameID;
+
+    auto completeManipulationForItems = [&](const Vector<WebCore::TextManipulationItem>& items) -> std::optional<Vector<TextManipulationControllerManipulationFailure>> {
+        ASSERT(!items.isEmpty());
+        auto* frame = WebProcess::singleton().webFrame(currentFrameID);
```

Comment:
(Same question about deploying `RefPtr` here)

Reply:
I don't think we need to use RefPtr here since all we're doing with this pointer is to get coreFrame, which is really a trivial function.